docs/89-NEWS.md

```diff
@@ -0,0 +1,839 @@
+fixed.
+**See:**
+
+* `gears.shape`
+
+#### The hierarchy API
+
+The widget framework now produces a persistent model of its content rather than
+a volatile one during the wibox drawing. This allows for better introspection
+into the widget tree.
+This model is now exposed through the `mouse::enter`, `mouse::press`,
+`mouse.current_widget` and other APIs. This tree model also includes various
+matrices to convert positions from the screen coordinates to the one of the
+widgets (think of scaling and rotations).
+
+This allows for interactive widgets such as a slider.
+
+**See:**
+
+* `gears.hierarchy`
+* * <a href="./04-new-widgets.md.html">Creating new widget</a>
+
+### New themes
+
+A new `xresources` theme has been added. It uses native X11 assets such as
+colors.
+
+### Spawn and launcher improvements
+
+#### Spawn
+
+The newly renamed `awful.spawn` (previously `awful.util.spawn`) has been
+extended into a whole API. It is now possible to define rules directly through
+the spawn function.
+Note that this only works if the client properly supports the freedesktop.org
+startup notifications protocol. For example, to open a new `urxvt` in a new tag
+from the command line, use:
+
+
+    awesome-client  "require('awful.spawn')('urxvt', {new_tag=true})"
+
+
+As another example, to launch a centered floating terminal in the currently
+selected tag of screen number 2:
+
+
+    awful.spawn("urxvt", {
+        tag       = screen[2].selected_tag,
+        placement = awful.placement.centered,
+        floating  = true
+    })
+
+**See:**
+
+* `awful.spawn`
+* `awful.rules`
+
+All rule properties can be used, including the newly introduced placement ones
+(like above).
+
+#### Launcher
+
+`awful.prompt` gained many new features. One of them is the ability to add
+custom keyboard shortcuts and mutate the command.
+Paired with the new spawn features, it can be used to create `mod4 + r`
+shortcuts to spawn the clients with arguments and callbacks.
+
+**See:**
+
+* `awful.prompt`
+
+### Focus stealing filters.
+
+It is now possible to add and manage filters to restrict what kind of focus
+stealing is allowed. It can be used to mute noisy applications or to implement
+tag level policies.
+Every way a client could claim focus, including those from within Awesome
+itself, now goes through the request filters.
+
+**See:**
+
+* `awful.ewmh.add_activate_filter`
+* `awful.ewmh.remove_activate_filter`
+* `awful.ewmh.activate`
+
+### Notification actions
+
+Awesome now supports XDG notification (aka, naughty) actions.
+
+**See:**
+
+* `naughty`
+
+### Custom xproperties support
+
+Awesome can now save some data in the X11 server itself. This allows to
+communicate with external applications or so save state across restarts.
+
+There is also supported/used for persistent client properties.
+
+### Better XKB keyboard layout support
+
+Awesome now has native support for keyboard layout detection and setting. Using
```

Comment:
Can we really set the keyboard layout? I think we can only set the active "group" of the layout. So in fact `setxkbmap` is still needed.